utils/src/main/kotlin/Extensions.kt

```diff
@@ -47,6 +47,12 @@ import java.security.MessageDigest
  */
 fun ByteArray.toHexString(): String = joinToString("") { String.format("%02x", it) }
 
+/**
+ * Return the duplicates as identified by [keySelector] of a collection.
+ */
+fun <T, K> Collection<T>.getDuplicates(keySelector: (T) -> K): Set<K> =
```

Comment:
Given the way we use this function - check that there are no duplicates -, it may even be beneficial to have an additional function like
`fun <T,K> Collection<T>.requireNoDuplicates(errMsg: String, keySelector: (T) -> K)`

Reply:
For now, that's still a tad too specific for my taste... but I agree it's an idea to keep in mind.